plugins/modules/certificate.py

```diff
@@ -0,0 +1,151 @@
+    type: str
+    required: true
+notes:
+  - C(check_mode) is not supported.
+"""
+
+EXAMPLES = r"""
+- name: Upload new certificate
+  scale_computing.hypercore.certificate:
+    private_key: "{{ lookup('file', 'private_key.key') }}"
+    certificate: "{{ lookup('file', 'scale_cert.cer') }}"
+"""
+
+RETURN = r"""
+record:
+  description:
+    - Certificate record.
+  returned: success
+  type: dict
+  sample:
+    private_key: this_is_the_private_key
+    certificate: this_is_the_certificate
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors
+from ..module_utils.utils import is_changed
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.typed_classes import (
+    TypedDiff,
+    TypedTaskTag,
+)
+from ..module_utils.task_tag import TaskTag
+
+from typing import Union, Tuple
+import ssl
+from time import sleep
+
+
+def get_certificate(module: AnsibleModule) -> str:
+    host = (
+        module.params["cluster_instance"]["host"]
+        .replace("https://", "")
+        .replace("http://", "")
+    )
+    cert = ssl.get_server_certificate((host, 443))
+    return cert
+
+
+def upload_cert(module: AnsibleModule, rest_client: RestClient) -> TypedTaskTag:
+    payload = dict(
+        certificate=module.params["certificate"],
+        privateKey=module.params["private_key"],
+    )
+    response = rest_client.create_record("/rest/v1/Certificate", payload, False)
+    return response
+
+
+def ensure_present(
+    module: AnsibleModule,
+    rest_client: RestClient,
+) -> Tuple[bool, Union[str, None], TypedDiff]:
+    before = get_certificate(module)
+    if before.replace("\n", "") == module.params["certificate"].replace("\n", ""):
+        return False, None, dict(before=None, after=None)
+    else:
+        task = upload_cert(module, rest_client)
+        max_retries = 10
+        for ii in range(max_retries):
+            try:
+                TaskTag.wait_task(rest_client, task)
+                break
+            except (
+                errors.ScaleComputingError
+            ) as ex:  # ConnectionRefusedError not working
+                if str(ex) == "[Errno 111] Connection refused":
+                    sleep(2)
+                    continue
+                else:
+                    raise
+        after = get_certificate(module)
+    return is_changed(before, after), after, dict(before=before, after=after)
+
+
+def run(
+    module: AnsibleModule, rest_client: RestClient
+) -> Tuple[bool, Union[str, None], TypedDiff]:
+    return ensure_present(module, rest_client)
+
+
+def main() -> None:
+    module = AnsibleModule(
+        supports_check_mode=False,
+        argument_spec=dict(
+            arguments.get_spec("cluster_instance"),
+            private_key=dict(
+                type="str",
+                required=True,
+                no_log=True,
+            ),
+            certificate=dict(
```

Comment:
Right now this is `NO_LOG=FALSE,` since we output certificate as string as Ansible output.
If this is something that should be hidden, we can change output to be something like `"Certificate was uploaded"` or `"Certificate is already in use"`.

Reply:
Agreed, `certificate` is public part (we need to hide only `private_key`).